follow exception 1.12.2

Error when i trying follow entity

### Error log
`[20:45:00] [Client thread/INFO] [net.minecraft.client.gui.GuiNewChat]: [CHAT] [Baritone] > follow entity enderman
[20:45:00] [Client thread/INFO] [net.minecraft.client.gui.GuiNewChat]: [CHAT] [Baritone] An unhandled exception occurred.The error is in your game's log, please report this at https://github.com/cabaletta/baritone/issues
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: baritone.api.command.exception.CommandUnhandledException: java.lang.NoSuchFieldError: field_191308_b
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at baritone.cc.execute(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at baritone.x.a(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at baritone.x.onSendChatMessage(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at baritone.ce.a(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at baritone.ce$$Lambda$881/1452293471.accept(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at baritone.ce.onSendChatMessage(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.entity.EntityPlayerSP.handler$sendChatMessage$zeg000(EntityPlayerSP.java)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.entity.EntityPlayerSP.func_71165_d(EntityPlayerSP.java)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.gui.GuiScreen.func_175281_b(GuiScreen.java:424)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.gui.GuiScreen.func_175275_f(GuiScreen.java:411)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.gui.GuiChat.func_73869_a(SourceFile:81)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.gui.GuiScreen.func_146282_l(GuiScreen.java:558)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:512)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1759)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1098)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:398)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.client.main.Main.main(SourceFile:123)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at java.lang.reflect.Method.invoke(Method.java:497)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
[20:45:00] [Client thread/INFO] [STDERR]: [baritone.api.command.exception.CommandUnhandledException:handle:-1]: 	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
[20:45:00] [Client thread/INFO] [STDERR]: [java.lang.Throwable:printStackTrace:634]: Caused by: java.lang.NoSuchFieldError: field_191308_b
[20:45:00] [Client thread/INFO] [STDERR]: [java.lang.Throwable:printStackTrace:634]: 	at baritone.api.command.datatypes.EntityClassById.get(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [java.lang.Throwable:printStackTrace:634]: 	at baritone.api.command.datatypes.EntityClassById.get(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [java.lang.Throwable:printStackTrace:634]: 	at baritone.ag.getDatatypeFor(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [java.lang.Throwable:printStackTrace:634]: 	at baritone.aw.execute(Unknown Source)
[20:45:00] [Client thread/INFO] [STDERR]: [java.lang.Throwable:printStackTrace:634]: 	... 24 more`

--- a/src/api/java/baritone/api/command/datatypes/EntityClassById.java
+++ b/src/api/java/baritone/api/command/datatypes/EntityClassById.java
@@ -32,7 +32,19 @@ public enum EntityClassById implements IDatatypeFor<Class<? extends Entity>> {
     public Class<? extends Entity> get(IDatatypeContext ctx) throws CommandException {
         ResourceLocation id = new ResourceLocation(ctx.getConsumer().getString());
         Class<? extends Entity> entity;
-        if ((entity = EntityList.REGISTRY.getObject(id)) == null) {
+        try {
+            entity = EntityList.REGISTRY.getObject(id);
+        } catch(NoSuchFieldError e) {
+            // Forge removes EntityList.REGISTRY field and provides the getClass method as a replacement
+            // See https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/patches/minecraft/net/minecraft/entity/EntityList.java.patch
+            try {
+                entity = (Class<? extends Entity>) EntityList.class.getMethod("getClass", ResourceLocation.class).invoke(null, id);
+            } catch (Exception ex) {
+                throw new RuntimeException("EntityList.REGISTRY does not exist and failed to call the Forge-replacement method", ex);
+            }
+        }
+
+        if (entity == null) {
             throw new IllegalArgumentException("no entity found by that id");
         }
         return entity;

